portfolio-project-matching-app/backend/dao.js

```diff
@@ -356,28 +378,51 @@ const incrementProjectCensusAndClose = async (coll, id) => {
     READ
 */
 
-// see daoProject.js, daoUser.js, and daoTechnology.js
-
-const readAllDocs = async (coll) => {
+const readAllObjects = async (coll, deep = false) => {
     /*
     DESCRIPTION:    retrieves all documents in the specified collection and
-                    returns an array of custom objects. Note, the associations
-                    will not be populated in the objects.
+                    returns an array of custom objects.
 
     INPUT:          coll (string): collection to get documents from
 
-    RETURN:         array of custom objects
+                    deep (boolean): indicates whether objects association
+                    properties should be populated. If no argument is entered,
+                    returned objects will be shallow (no association arrays
+                    populated)
+
+    RETURN:         array of custom objects ([Project], [User], etc)
```

Comment:
Since the function returns custom objects, I altered the name a bit. Also, I utilize helper functions in branching depending on the type of object that should be returned and whether the objects should be "deep" (association arrays populated with shallow objects).